includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -879,6 +879,26 @@ public function update_fees( $order, $balance_transaction_id ) {
 		}
 	}
 
+	/**
+	 * Updates Stripe currency in order meta.
+	 *
+	 * @since 4.1.7
+	 * @param object $order The order object
+	 * @param int $balance_transaction_id
+	 */
+	public function update_currency( $order, $balance_transaction_id ) {
+		$order_id = WC_Stripe_Helper::is_pre_30() ? $order->id : $order->get_id();
+
+		$balance_transaction = WC_Stripe_API::retrieve( 'balance/history/' . $balance_transaction_id );
+
+		if ( empty( $balance_transaction->error ) ) {
+			$currency = ! empty( $balance_transaction->currency ) ? strtoupper( $balance_transaction->currency ) : null;
+			WC_Stripe_Helper::update_stripe_currency( $order, $currency );
```

Comment:
What happens if we set the currency to null? Should we update the conditional to branch here to `empty( error ) && ! empty( currency )`?

Reply:
Makes sense.